commit a13271f2feb6e480b2e698d4efa3b94150a6808b
Author: Tapani Pälli <tapani.palli@intel.com>
Date:   Tue Nov 28 09:23:29 2017 +0200

    glx: do not pick sRGB config for 32-bit RGBA visual
    
    This fixes blending issues seen with kwin and gnome-shell when
    32bit visual has sRGB capability set.
    
    Reviewed-by: Adam Jackson <ajax@redhat.com>
    Signed-off-by: Tapani Pälli <tapani.palli@intel.com>
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=103699
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=103646
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=103655
    (cherry picked from commit c2954b16c8730c7ed8441fd8dba25900f3aed265)

diff --git a/glx/glxscreens.c b/glx/glxscreens.c
index 29bacd9..62d66bc 100644
--- a/glx/glxscreens.c
+++ b/glx/glxscreens.c
@@ -275,6 +275,11 @@ pickFBConfig(__GLXscreen * pGlxScreen, VisualPtr visual)
         /* If it's the 32-bit RGBA visual, demand a 32-bit fbconfig. */
         if (visual->nplanes == 32 && config->rgbBits != 32)
             continue;
+        /* If it's the 32-bit RGBA visual, do not pick sRGB capable config.
+         * This can cause issues with compositors that are not sRGB aware.
+         */
+        if (visual->nplanes == 32 && config->sRGBCapable == GL_TRUE)
+            continue;
         /* Can't use the same FBconfig for multiple X visuals.  I think. */
         if (config->visualID != 0)
             continue;
